aspnet/fundamentals/logging.rst

```diff
@@ -126,105 +180,251 @@ Debug          dbug
+        => RequestId:0HKV9C49II9CK RequestPath:/api/todo/0 => TodoApi.Controllers.TodoController.GetById (TodoApi) => This message is attached to all logs written within the using block
+        Getting item 0
+  warn: TodoApi.Controllers.TodoController[4000]
+        => RequestId:0HKV9C49II9CK RequestPath:/api/todo/0 => TodoApi.Controllers.TodoController.GetById (TodoApi) => This message is attached to all logs written within the using block
+        GetById(0) NOT FOUND 
 
-The yellow line with the "warn: " prefix, along with the following line, is output by the ``ConsoleLogger``. The next line, beginning with "TodoApi.Controllers.TodoController", is output from the TraceSource logger. There are many other TraceSource listeners available, and the ``TextWriterTraceListener`` can be configured to use any ``TextWriter`` instance, making this a very flexible option for logging.
+TraceSource Logging
+-------------------
 
-Configuring Other Providers
-^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Applications that run on the full .NET Framework can use the `System.Diagnostics.TraceSource <https://msdn.microsoft.com/library/system.diagnostics.tracesource.aspx>`_ libraries and providers. ``TraceSource`` allows you to route messages to a variety of `listeners <https://msdn.microsoft.com/library/4y5y10s7.aspx>`__, including the `EventLogTraceListener <https://msdn.microsoft.com/library/system.diagnostics.eventlogtracelistener.aspx>`__ and the `EventProviderTraceListener <https://msdn.microsoft.com/library/system.diagnostics.eventing.eventprovidertracelistener.aspx>`__ for `ETW tracing <https://msdn.microsoft.com/library/ms751538.aspx>`__.
 
-In addition to the built-in loggers, you can configure logging to use other providers. Add the appropriate package to your *project.json* file, and then configure it just like any other provider. Typically, these packages include extension methods on ``ILoggerFactory`` to make it easy to add them.
+To use ``TraceSource``, add the ``Microsoft.Extensions.Logging.TraceSource`` package to your project, along with any specific trace source packages you'll be using, such as ``TextWriterTraceListener`` in the sample application that goes with this article.
+
+The following example demonstrates how to configure a ``TraceSourceLogger`` instance for an application, with a listener that processes only ``Warning`` or higher priority messages. Each call to ``AddTraceSource`` takes a ``TraceListener``. The code shown configures a ``TextWriterTraceListener`` to write to the console window.
+
+.. literalinclude:: logging/sample/src/TodoApi/Startup.cs
+  :language: c#
+  :start-after: snippet_TraceSource
+  :end-before: endregion
+  :emphasize-lines: 8-12
+  :dedent: 8
 
- * `elmah.io <https://github.com/elmahio/Elmah.Io.Extensions.Logging>`_ - provider for the elmah.io service
+The ``sourceSwitch`` is configured to use ``SourceLevels.Warning``, so only ``Warning`` (or higher) log messages are picked up by the ``TraceListener`` instance.
```

Comment:
~~The ``sourceSwitch`` is configured to use ``SourceLevels.Warning``, so only ``Warning`` (or higher) log messages are picked up by the ``TraceListener`` instance.~~